contrib/openssl/async_engine.c

```diff
@@ -274,7 +294,7 @@ static int async_rsa_priv_enc(int flen, const unsigned char *from,
 static int async_rsa_priv_dec(int flen, const unsigned char *from,
                       unsigned char *to, RSA *rsa, int padding)
 {
-    //printf("async_priv_dec\n");
+    printf("async_priv_dec\n");
     spawn_delay_thread();
     async_pause_job();
```

Comment:
It is required to clear the wake signal by `read(pipefds[0], &buf, 1);`.

Reply:
That clearing is actually done within the async_pause_job function.  Rather after the return from the openssl function ASYNC_pause_job

Reply:
Sorry, I confused async_pause_job with ASYNC_pause_job.